syntax = "proto3";

package fraud;

// Our new gRPC service name
service FraudDetectionService {
    // The method the Orchestrator will call
    rpc CheckOrder (CheckOrderRequest) returns (CheckOrderResponse);
}

// The request message
message CheckOrderRequest {
    // For this demo, let's assume we only need totalAmount to decide.
    // You can add user data, item list, etc. as fields as needed.
    double totalAmount = 1;
}

// The response message
message CheckOrderResponse {
    bool isFraud = 1;
    string reason = 2;
}
